### ISSUE TITLE
copySync('my_symlink') always dereferences the symlink 

### ISSUE THREAD

**Issue by maikokuppe** [jprichardson/node-fs-extra#447]
Given this folder structure

```
file.txt
my_symlink -> file.txt
```

running one of these commands

```
fs.copySync('my_symlink', 'my_symlink_copy')
fs.copySync('my_symlink', 'my_symlink_copy', { dereference: false })
fs.copySync('my_symlink', 'my_symlink_copy', { dereference: true })
```

yields

```
file.txt
my_symlink -> file.txt
my_symlink_copy // this is a hardlink
```

This workaround works though:

```
fs.copySync('my_symlink', 'my_symlink_copy', { recursive: true })
```

**Comment by manidlou** [comment]
@maikokuppe `recursive` option has never been documented. In fact, I brought it up in #424. When `recursive` option is removed from the `copySync()` source code, your first command will most likely return the expected result.

**Comment by manidlou** [comment]
Fixed in 5.0.0 :tada:
